src/categorical_algebra/DataMigrations.jl

```diff
@@ -41,102 +41,107 @@ a "duc query" (disjoint union of conjunctive queries).
 """
 const GlucQuery{C<:FinCat} = Diagram{id,<:TypeCat{<:Diagram{op,C}}}
 
-""" Functor defining a pullback or delta data migration.
+""" A data migration is guaranteed to contain a functor between
+schemas that can be used to migrate data between acsets
+on those schemas. This functor ``F`` may be partial, in which case
+it contains extra data describing how to use it to execute a migration. The
+migration may be a pullback data migration ([`DeltaMigration`](@ref)), specified by a
+functor ``D → C`` between the schemas, or a conjunctive, gluing,
+duc, or gluc (contravariant) data migration. Alternatively, ``F`` may
+be a schema functor specifying a ``Σ`` migration in the covariant direction.
 """
-const DeltaSchemaMigration{D<:FinCat,C<:FinCat} = FinFunctor{D,C}
+abstract type AbstractDataMigration{F<:FinDomFunctor} end
+func(M::AbstractDataMigration) = M.functor
+abstract type AbstractContravariantMigration{F<:FinDomFunctor} <: AbstractDataMigration{F} end
+abstract type AbstractCovariantMigration{F<:FinDomFunctor} <: AbstractDataMigration{F} end
```

Comment:
We don't have to prefix all abstract types with `Abstract`. I'd mildly prefer to shorten these to `ContravariantMigration` and `CovariantMigration`.